bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -0,0 +1,383 @@
+        }
+        bulbMode = Helper.resolveJSON(messageJSON, "\"bulb_mode\":\"", 5);
+        switch (bulbMode) {
+            case "white":
+                if (!globeType.equals("cct") && !globeType.equals("fut091")) {
+                    changeChannel(CHANNEL_BULB_MODE, new StringType("white"));
+                    changeChannel(CHANNEL_COLOUR, new HSBType("0,0," + tempBulbLevel));
+                    changeChannel(CHANNEL_DISCO_MODE, new StringType("None"));
+                }
+                String bulbCTemp = Helper.resolveJSON(messageJSON, "\"color_temp\":", 3);
+                if (!bulbCTemp.isEmpty()) {
+                    int ibulbCTemp = (int) Math.round(((Float.valueOf(bulbCTemp) / 2.17) - 171) * -1);
+                    changeChannel(CHANNEL_COLOURTEMP, new PercentType(ibulbCTemp));
+                }
+                break;
+            case "color":
+                changeChannel(CHANNEL_BULB_MODE, new StringType("color"));
+                changeChannel(CHANNEL_DISCO_MODE, new StringType("None"));
+                String bulbHue = Helper.resolveJSON(messageJSON, "\"hue\":", 3);
+                String bulbSaturation = Helper.resolveJSON(messageJSON, "\"saturation\":", 3);
+                if (bulbHue.isEmpty()) {
+                    logger.warn("Milight MQTT message came in as being a colour mode, but was missing a HUE value.");
+                } else {
+                    if (bulbSaturation.isEmpty()) {
+                        bulbSaturation = "100";
+                    }
+                    changeChannel(CHANNEL_COLOUR, new HSBType(bulbHue + "," + bulbSaturation + "," + tempBulbLevel));
+                }
+                break;
+            case "scene":
+                if (!globeType.equals("cct") && !globeType.equals("fut091")) {
+                    changeChannel(CHANNEL_BULB_MODE, new StringType("scene"));
+                }
+                String bulbDiscoMode = Helper.resolveJSON(messageJSON, "\"mode\":", 1);
+                if (!bulbDiscoMode.isEmpty()) {
+                    changeChannel(CHANNEL_DISCO_MODE, new StringType(bulbDiscoMode));
+                }
+                break;
+            case "night":
+                if (!globeType.equals("cct") && !globeType.equals("fut091")) {
+                    changeChannel(CHANNEL_BULB_MODE, new StringType("night"));
+                    if (config.oneTriggersNightMode) {
+                        changeChannel(CHANNEL_LEVEL, new PercentType("1"));
+                    }
+                }
+                break;
+        }
+    }
+
+    /*
+     * Used to calculate the colour temp for a globe if you want the light to get warmer as it is dimmed like a
+     * traditional halogen globe
+     */
+    private int autoColourTemp(int brightness) {
+        return minColourTemp.subtract(
+                minColourTemp.subtract(maxColourTemp).divide(new BigDecimal(100)).multiply(new BigDecimal(brightness)))
```

Comment:
Please make `new BigDecimal(100)` a constant so you can reuse it.

Reply:
DONE.